### PR TITLE
Add modules support, add changes from pipes-api

### PR DESCRIPTION
**Pull request by idexter**
- Added modules support.
- Added Changes from [toggl/pipes-api](https://github.com/toggl/pipes-api)

The reason I did this, is because I want to get rid of local dependency which for some reasons was introduced in [toggl/pipes-api](https://github.com/toggl/pipes-api/blob/5836abf482bf0eadeda7e32d4b48c8afee3dfdaf/go.mod#L5)

### CODE REVIEW

File: basecamp.go [hunk 1]
```diff
@@ -193,6 +202,10 @@ func (c *Client) get(url string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	fmt.Println("BASECAMP REQ", url, "time", time.Since(start))
```
Comment (myhro) [inline basecamp.go]: Do we need this message here? If so, can we use `log` instead?

File: basecamp.go [hunk 2]
```diff
@@ -193,6 +202,10 @@ func (c *Client) get(url string) ([]byte, error) {
...
+	if 304 == resp.StatusCode {
```
Comment (myhro) [inline basecamp.go]: Can we use status code constants from `net/http`? I'm also in favour avoiding [Yoda conditions][1].

[1]: https://en.wikipedia.org/wiki/Yoda_conditions

**Review by myhro — CHANGES_REQUESTED** [review]

File: basecamp.go (hunk 1 above)
Comment (idexter) [reply basecamp.go]: No, we don't. It's just was backported from `pipes-api`. I removed it

File: basecamp.go (hunk 2 above)
Comment (idexter) [reply basecamp.go]: Fixed

**Review by myhro — APPROVED** [review]
LGTM.

**Pull request merged**